code/weapon/weapons.cpp

```diff
@@ -3449,23 +3449,29 @@ int parse_weapon(int subtype, bool replace, const char *filename)
 	if(optional_string("$FOF:")){
 		stuff_float(&wip->field_of_fire);
 
-		if(optional_string("+FOF Spread Rate:")){
+		bool max_required = false;
+		if (optional_string("+FOF Spread Rate:")) {
 			stuff_float(&wip->fof_spread_rate);
-			if(required_string("+FOF Reset Rate:")){
+			if (required_string("+FOF Reset Rate:")) {
 				stuff_float(&wip->fof_reset_rate);
 			}
+			max_required = true;
+		}
 
-			if(required_string("+Max FOF:")){
-				float max_fof;
-				stuff_float(&max_fof);
-				wip->max_fof_spread = max_fof - wip->field_of_fire;
+		if (optional_string("+Max FOF:")){
+			float max_fof;
+			stuff_float(&max_fof);
+			wip->max_fof_spread = max_fof - wip->field_of_fire;
 
-				if (wip->max_fof_spread <= 0.0f) {
-					Warning(LOCATION, "WARNING: +Max FOF must be at least as big as $FOF for '%s'! Defaulting to match $FOF, no spread will occur!", wip->name);
-					wip->max_fof_spread = 0.0f;
-				}
+			if (wip->max_fof_spread <= 0.0f) {
+				Warning(LOCATION, "WARNING: +Max FOF must be at least as big as $FOF for '%s'! Defaulting to match $FOF, no spread will occur!", wip->name);
+				wip->max_fof_spread = 0.0f;
 			}
 		}
+
+		if (max_required && wip->max_fof_spread <= 0.0f) {
+			Error(LOCATION, "ERROR: +Max FOF for '%s' must be used if +FOF Spread Rate: is used!", wip->name);
```

Comment:
Instead of making this an `Error`, what about making it a `Warning` and setting the values back to factory defaults?

Also, use `error_display` (0 for warning, 1 for error) so that the message displays the parsed file and parsed line number.

Reply:
The only reason it's an error is because it was strictly required by the previous parsing, for posterity.

Reply:
Fair... the comment could be directed at the previous parsing code as well.